File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    static int[] visit=new int[201];
    public int solution(int n, int[][] computers) {
        int answer = 0;    
        for(int i=0; i<n; i++){
            if (visit[i]==0){
                BFS(i,n,computers);
                answer+=1;
            }
            
        }
        return answer;
    }
    public void BFS(int x,int n, int[][] computers){
        Queue<Integer> dq=new LinkedList<>();
        dq.add(x);
        visit[x]=1;
        while (!dq.isEmpty()){
            int k=dq.poll();
            for(int i=0; i<n; i++){
                if (computers[k][i]==1 && visit[i]==0){
                    visit[i]=1;
                    dq.add(i);
                        
                }
            }
        }
        
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):            
    graph=[[]*(n+1) for _ in range(n+1)]
    for i in range(n):
        for j in range(n):
            if i==j:
                continue
            if computers[i][j]==1:
                graph[i+1].append((j+1))
                graph[j+1].append((i+1))
        
    ans=0
    visit=[0]*(n+1)
    def BFS(x):
        dq=deque()
        dq.append(x)
        visit[x]=1    
        while dq:
            x=dq.popleft()
            for next_node in graph[x]:
                if not visit[next_node]:
                    visit[next_node]=1    
                    dq.append(next_node)
    for i in range(1, n+1):
        if not visit[i]:
            BFS(i)
            ans+=1
    return ans